### PR TITLE
feat #4448: add support for JWT tokens

### PR DESCRIPTION
**Pull request by dgnorton**
###### Required for all non-trivial PRs
- [x] Rebased/mergable
- [x] Tests pass
- [x] CHANGELOG.md updated

### CODE REVIEW

File: services/httpd/handler.go [hunk 1]
```diff
@@ -615,21 +618,53 @@ func resultError(w http.ResponseWriter, result influxql.Result, code int) {
...
+// As params: http://127.0.0.1/query?u=username&p=password
+// As basic auth: http://username:password@127.0.0.1
+// As Bearer token in Authorization header: Bearer <JWT_TOKEN_BLOB>
+func parseCredentials(r *http.Request) (*credentials, error) {
 	q := r.URL.Query()
 
-	if u, p := q.Get("u"), q.Get("p"); u != "" && p != "" {
-		return u, p, nil
+	// Check for the HTTP Authorization header.
+	if s := r.Header.Get("Authorization"); s != "" {
```
Comment (markbates) [inline services/httpd/handler.go]: This should probably be `X-Authorization`

Comment (jwilder) [inline services/httpd/handler.go]: What about Authorization: Bearer <token>?

Comment (dgnorton) [inline services/httpd/handler.go]: As I understand it, the `Authorization` header allows for custom schemes.

OAuth2 bearer tokens:  https://tools.ietf.org/html/rfc6750#page-5

AWS scheme:  http://docs.aws.amazon.com/AmazonS3/latest/dev/RESTAuthentication.html#ConstructingTheAuthenticationHeader

Comment (markbates) [inline services/httpd/handler.go]: Never mind, I didn't realize `Authorization` was a standard header these days. Typically when using custom headers, you always prepend them with `X-` first.

File: services/httpd/handler.go [hunk 2]
```diff
@@ -615,21 +618,53 @@ func resultError(w http.ResponseWriter, result influxql.Result, code int) {
...
+	if u, p := q.Get("u"), q.Get("p"); u != "" && p != "" {
+		return &credentials{
+			Method:   UserAuthentication,
+			Username: u,
+			Password: p,
+		}, nil
 	}
-	return "", "", fmt.Errorf("unable to parse Basic Auth credentials")
+
+	return nil, fmt.Errorf("unable to parse Basic Auth credentials")
```
Comment (e-dard) [inline services/httpd/handler.go]: I think this is more general than `Basic Auth` now. Maybe `unable to parse authentication credentials`?

**Comment by e-dard** [comment]
Couple of nits but LGTM 👍

File: services/httpd/handler.go (hunk 1 above)
Comment (simi) [inline services/httpd/handler.go]: @markbates http://www.rfc-editor.org/info/rfc6648

**Comment by retorquere** [comment]
Is there documentation on how to use this in InfluxDB?

**Pull request merged**